capstone/src/main/webapp/sheetsScript.js

```diff
@@ -62,29 +64,272 @@ async function gatherSheetsInput() {
     }
   }
 
-  const text = await getTextFromSheet(id);
-  handleInput(text, langElement.value, requestedAttributes, tokenizer);
+  // Show general output
+  let sheetNames = await getSheetNames(id);
+  let totalText = '';
+  let range = '!A1:YY';
+  for (const name of sheetNames) {
+    const sheet = await getSpreadsheet(id, name, range);
+    const text = await getTextFromSheet(sheet);
+    totalText += text + '\n';
+  }
+  handleInput(totalText, langElement.value, requestedAttributes, tokenizer);
+
+  // Create spreadsheet if requested
+  const userDecisionElement = document.getElementById('sheets-output-yes-no');
+  if (userDecisionElement != null && userDecisionElement.value === 'yes') {
+    let title = 'Perspective Output';
+    const titleElement = document.getElementById('sheets-title');
+    if (titleElement != null && titleElement.value != '') {
+      title = titleElement.value;
+    }
+
+    const newSheetId = await createSheet(title);
+
+    // Check if user has entered a custom range
+    const userRangeElement = document.getElementById('sheets-range-yes-no');
+    if (userRangeElement != null && userRangeElement.value === 'yes') {
+      // Check if this is a valid Sheets range
+      const rangeElement = document.getElementById('sheets-range');
+      if (rangeElement != null) {
+        const rangeRegEx = rangeElement.value.match(/(.+)(![\w]*[\d]*(:[\w]*[\d]*)?)/);
```

Comment:
nit: Could we put a comment for this regexp?

Reply:
Will do.